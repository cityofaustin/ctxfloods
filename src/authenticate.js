const authenticate = function(lokka, userType, email, password) {
  describe('When signing in as a ' + userType, () => {
    describe('with the correct username and password', () => {
      it('should generate a valid jwtToken', async () => {
        const response = await lokka.send(`
          mutation($email:String!, $password:String!) {
            authenticate(input: {email: $email, password: $password}) {
              jwtToken
            }
          }
        `,
        {
          email: email,
          password: password
        });

        expect(response.authenticate.jwtToken).not.toBeNull();
      });
    });

    describe('with an incorrect username and password', () => {
      it('should generate a null jwtToken', async () => {
        const response = await lokka.send(`
          mutation($email:String!, $password:String!) {
            authenticate(input: {email: $email, password: $password}) {
              jwtToken
            }
          }
        `,
        {
          email: email,
          password: "wrong"
        });

        expect(response.authenticate.jwtToken).toBeNull();
      });
    });
  });
}

export default authenticate;
