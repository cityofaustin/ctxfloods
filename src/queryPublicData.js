// TODO: Figure out how to use this in the community editor/admin/super admin tests
const queryPublicData = function(lokka) {
  describe('When querying public data', () => {

    it('should get all users', async () => {
      const response = await lokka.send(`
      {
        allUsers {
          nodes {
            id
          }
        }
      }
      `);

      expect(response).not.toBeNull();
    });

    it('should get all communities', async () => {
      const response = await lokka.send(`
      {
        allCommunities {
          nodes {
            id
          }
        }
      }
      `);

      expect(response).not.toBeNull();
    });

    it('should get all statuses', async () => {
      const response = await lokka.send(`
      {
        allStatuses {
          nodes {
            id
          }
        }
      }
      `);

      expect(response).not.toBeNull();
    });

    it('should get all status updates', async () => {
      const response = await lokka.send(`
      {
        allStatusUpdates {
          nodes {
            id
          }
        }
      }
      `);

      expect(response).not.toBeNull();
    });

    it('should get all status reasons', async () => {
      const response = await lokka.send(`
      {
        allStatusReasons {
          nodes {
            id
          }
        }
      }
      `);

      expect(response).not.toBeNull();
    });

    it('should get all status durations', async () => {
      const response = await lokka.send(`
      {
        allStatusDurations {
          nodes {
            id
          }
        }
      }
      `);

      expect(response).not.toBeNull();
    });

    it('should get all crossings', async () => {
      const response = await lokka.send(`
      {
        allCrossings {
          nodes {
            id
          }
        }
      }
      `);

      expect(response).not.toBeNull();
    });

    it('should get all community crossings', async () => {
      const response = await lokka.send(`
      {
        allCommunityCrossings {
          nodes {
            id
          }
        }
      }
      `);

      expect(response).not.toBeNull();
    });

    it('should get the latest statuses correctly', async () => {
      const response = await lokka.send(`
        {
          allCrossings {
            nodes {
              latestStatus {
                id
              }
            }
          }
        }
      `);

      expect(response).not.toBeNull();
    });

  });
}

export default queryPublicData;
