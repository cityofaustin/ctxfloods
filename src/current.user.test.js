import HttpTransport from 'lokka-transport-http';
import Lokka from 'lokka';
import queryPublicData from './queryPublicData';
import authenticate from './authenticate';

const anonLokka = new Lokka({transport: new HttpTransport('http://localhost:5000/graphql')});
const superAdminEmail = 'superadmin@flo.ods';
const communityAdminEmail = 'admin@community.floods';
const communityEditorEmail = 'editor@community.floods';
const everyPassword = 'texasfloods';
const wrongPassword = 'wrong';

async function getToken(email, password) {
  const response = await anonLokka.send(`
    mutation($email:String!, $password:String!) {
      authenticate(input: {email: $email, password: $password}) {
        jwtToken
      }
    }
  `,
  {
    email: email,
    password: password
  });

  return response.authenticate.jwtToken;
}

function shouldWork(email, password, extra_description) {
  describe('as ' + email + ' ' + (extra_description || ''), () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(email, password).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport('http://localhost:5000/graphql', {headers})});
        done();
      });
    });

    it('should get the correct current user', async () => {
      const response = await lokka.send(`
        {
          currentUser {
            id
          }
        }
      `);

      expect(response).toMatchSnapshot();
    });
  }); 
}

function shouldFail(email, password, extra_description) {
      console.log("BKARG")
      console.log(email & password)

  describe('as ' + email + ' ' + (extra_description || ''), () => {  
    var lokka;

    beforeAll(async (done) => {

      if(!(email & password)) {
        console.log("ANON");
        lokka = anonLokka;
        done();
      }

      getToken(email, password).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport('http://localhost:5000/graphql', {headers})});
        done();
      });
    });

    it('should fail to get the current user', async () => {
      try {
        const response = await anonLokka.send(`
        {
          currentUser {
            id
          }
        }
      `);
      } catch(e) {
        expect(e).toMatchSnapshot();
      }
    });
  });
}

describe('When getting current user', () => {
  // shouldFail();
  shouldWork(superAdminEmail, everyPassword);
  //shouldFail(superAdminEmail, wrongPassword);
  shouldWork(communityAdminEmail, everyPassword);
  //shouldFail(communityAdminEmail, wrongPassword);
  shouldWork(communityEditorEmail, everyPassword);
  //shouldFail(communityEditorEmail, wrongPassword);
});
