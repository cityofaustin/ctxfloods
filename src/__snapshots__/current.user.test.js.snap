// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When getting current user as admin@community.floods  should get the correct current user 1`] = `
Object {
  "currentUser": Object {
    "id": 2,
  },
}
`;

exports[`When getting current user as editor@community.floods  should get the correct current user 1`] = `
Object {
  "currentUser": Object {
    "id": 3,
  },
}
`;

exports[`When getting current user as superadmin@flo.ods  should get the correct current user 1`] = `
Object {
  "currentUser": Object {
    "id": 1,
  },
}
`;
